myuw/views/api/textbook.py

```diff
@@ -34,27 +35,52 @@ def GET(self, request, year, quarter, summer_term):
     def respond(self, request, year, quarter, summer_term):
         timer = Timer()
         try:
-            if not is_student():
-                log_msg(logger, timer, "Not a student, abort!")
-                return data_not_found()
-
             term = get_specific_term(year=year, quarter=quarter)
-            schedule = get_schedule_by_term(request, term)
+            try:
+                schedule = get_schedule_by_term(request, term)
+            except DataFailureException as ex:
+                if ex.status == 404:
+                    schedule = None
+                    by_sln = {}
+                else:
+                    raise
+
+            # enrolled sections
+            if schedule:
+                if summer_term is not None and len(summer_term) > 0:
+                    summer_term = summer_term.replace(",", "")
+                    filter_schedule_sections_by_summer_term(
+                        schedule, summer_term)
 
-            if summer_term is not None and len(summer_term) > 0:
-                summer_term = summer_term.replace(",", "")
-                filter_schedule_sections_by_summer_term(schedule, summer_term)
+                if len(schedule.sections) == 0:
```

Comment:
If someone's an instructor and student, signed up for summer B term, it looks like if this is loaded during A term they'll get data_not_found, before it looks for their instructor book list.